### PR TITLE
Deeptiman - solution

### PR DESCRIPTION
**Pull request by codecrook**
My solution to the task in "Real Software Dev Squad" first-task.
, Max Movies in Year

### CODE REVIEW

**Comment by ankushdharkar** [comment]
@codecrook Solution is incorrect. Please check your usecases again

**Comment by codecrook** [comment]
Hi @ankushdharkar ! 
Updated the solution. Please review again.

File: maxMoviesInYear.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/**
+ * @name maxMoviesInYear
+ * @description Function to find years with maximum number of movies licensed
+ * @param {Array} moviesList The list of movies
+ * @returns {Number[]} Years with max movies on platform
+ */
+function maxMoviesInYears(moviesList) {
+    if (moviesList.length === 0) return [];
```
Comment (ankushdharkar) [inline maxMoviesInYear.js]: Do we need this line?

The more logic you don't  branch off, the better you prevent the cognitive overload

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

**Comment by ankushdharkar** [comment]
@codecrook Your logic works. But readability of your code is low

Let's break it into easy to understand functions. The smaller the parts you have to reason about, the more confidence in the code working and easier to maintain over long-term

```js
function maxMoviesInYears(moviesList) {
    const years = getYearsFromList(moviesList);
    const maxMovieYears = getMaxMovieYearsFromYears(years);
    return [...maxMovieYears];
}
```

**Comment by ankushdharkar** [comment]
@codecrook What is the time complexity and space complexity of your solution?

**Comment by codecrook** [comment]
Hi @ankushdharkar ,
Made the code changes as per review comments.
I understand time-complexity is a big problem here. Will improve on the course of my learning here.🙂

**Comment by codecrook** [comment]
Hi @ankushdharkar !
Added the time and space complexity calculations.
Please review.

File: README.md [hunk 2]
```diff
@@ -21,3 +21,61 @@ Imagine that we can fetch from an API the details of all movies that were ever s
...
+
+``` js
+function getMaxMovieYearsFromYears(years)
+```
+Similarly, `forEach` will have `O(N)` and `filter` has another `O(N)` making it `O(N^2)`
+
+```js
+function maxMoviesInYear(moviesList)
+``` 
+Calling both the functions above one time each, it will be having `O(N^2) + O(N^2) = O(N^2)`
```
Comment (ankushdharkar) [inline README.md:42]: O(N<sup>2</sup>) seems bad. Can you improve please?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**